project-api/ballerina-projects/src/main/java/io/ballerina/projects/balo/BaloFiles.java

```diff
@@ -0,0 +1,200 @@
+
+import com.google.gson.Gson;
+import io.ballerina.projects.directory.DocumentData;
+import io.ballerina.projects.directory.ModuleData;
+import io.ballerina.projects.directory.PackageData;
+import io.ballerina.projects.model.BallerinaToml;
+import io.ballerina.projects.model.Package;
+import io.ballerina.projects.model.PackageJson;
+import io.ballerina.projects.utils.ProjectUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.ballerina.projects.utils.ProjectConstants.MODULES_ROOT;
+
+/**
+ * Contains a set of utility methods that create an in-memory representation of a Ballerina project using a balo.
+ *
+ * @since 2.0.0
+ */
+public class BaloFiles {
+    private static final PathMatcher matcher = FileSystems.getDefault().getPathMatcher("glob:**.bal");
+    private static Gson gson = new Gson();
+
+    private BaloFiles() {
+    }
+
+    public static PackageData loadPackageData(String baloPath) {
+        Path absBaloPath = validateBaloPath(baloPath);
+
+        URI zipURI = URI.create("jar:" + absBaloPath.toUri().toString());
+        try (FileSystem zipFileSystem = FileSystems.newFileSystem(zipURI, new HashMap<>())) {
+
+            // Load BallerinaToml object
+            Path packageJsonPathInBalo = zipFileSystem.getPath("package.json");
+            BallerinaToml ballerinaToml = loadBallerinaToml(packageJsonPathInBalo);
+
+            // Load default module
+            Path defaultModulePathInBalo = zipFileSystem.getPath(MODULES_ROOT, ballerinaToml.getPackage().getName());
+            ModuleData defaultModule = loadModule(defaultModulePathInBalo);
+
+            // load other modules
+            Path modulesPathInBalo = zipFileSystem.getPath(MODULES_ROOT);
+            List<ModuleData> otherModules = loadOtherModules(modulesPathInBalo, defaultModulePathInBalo);
+
+            return PackageData.from(absBaloPath, defaultModule, otherModules);
+        } catch (IOException e) {
+            throw new RuntimeException("cannot read balo:" + baloPath);
+        }
+    }
+
+    private static Path validateBaloPath(String baloPath) {
+        if (baloPath == null) {
+            throw new IllegalArgumentException("baloPath cannot be null");
+        }
+
+        Path absBaloPath = Paths.get(baloPath).toAbsolutePath();
+        if (!absBaloPath.toFile().canRead()) {
+            throw new RuntimeException("insufficient privileges to balo: " + absBaloPath);
+        }
+        if (!absBaloPath.toFile().exists()) {
+            throw new RuntimeException("balo does not exists: " + baloPath);
+        }
+
+        if (!absBaloPath.toString().endsWith(".balo")) {
```

Comment:
We can use a constant for ".balo". There is already one in ProjectDirConstants. We can copy it to ProjectConstants and use it.

Reply:
Added